docs-website/generateDocsDir.ts

```diff
@@ -291,37 +295,154 @@ function markdown_enable_specials(
+    contents.content += row;
   }
-  if (!accounted_for_in_sidebar(filepath)) {
-    throw new Error(
-      `File not accounted for in sidebar ${filepath} - try adding it to docs-website/sidebars.js`
-    );
+  contents.content += "\n\n";
+
+  // Full details
+  for (const release of releases_list.data) {
+    let body: string;
+    if (releaseNoteVersions.has(release.tag_name)) {
+      body = release.body;
+      body = markdown_sanitize_and_linkify(body);
+
+      // Redo the heading levels. First we find the min heading level, and then
+      // adjust the markdown headings so that the min heading is level 3.
+      const heading_regex = /^(#+)\s/gm;
+      const max_heading_level = Math.min(
+        3,
+        ...[...body.matchAll(heading_regex)].map((v) => v[1].length)
+      );
+      body = body.replace(
+        heading_regex,
+        `${"#".repeat(3 - max_heading_level)}$1 `
+      );
+    } else {
+      // Link to GitHub.
+      body = `View the [release notes](${release.html_url}) for ${release.name} on GitHub.`;
+    }
+
+    const info = `## [${release.name}](${
+      release.html_url
+    }) {#${make_link_anchor(release.tag_name)}}
+
+Released on ${pretty_format_date(release.created_at)} by [@${
+      release.author.login
+    }](${release.author.html_url}).
+
+${body}\n\n`;
+
+    contents.content += info;
   }
+
+  return contents;
+}
+
+function write_markdown_file(
+  contents: matter.GrayMatterFile<string>,
+  output_filepath: string
+): void {
+  const pathname = path.dirname(output_filepath);
+  fs.mkdirSync(pathname, { recursive: true });
+  fs.writeFileSync(output_filepath, contents.stringify(""));
 }
+
+(async function main() {
+  for (const filepath of markdown_files) {
+    // console.log("Processing:", filepath);
+    const contents_string = fs.readFileSync(`../${filepath}`).toString();
+    const contents = matter(contents_string);
+
+    markdown_guess_title(contents, filepath);
+    markdown_add_slug(contents, filepath);
+    markdown_add_edit_url(contents, filepath);
+    markdown_rewrite_urls(contents, filepath);
+    markdown_enable_specials(contents, filepath);
+    // console.log(contents);
+
+    const out_path = `${OUTPUT_DIRECTORY}/${filepath}`;
+    write_markdown_file(contents, out_path);
+  }
+
+  // Generate the releases history.
+  {
+    const contents = await generate_releases_markdown();
+    write_markdown_file(contents, `${OUTPUT_DIRECTORY}/releases.md`);
+    markdown_files.push("releases.md");
+  }
+
+  // Error if a doc is not accounted for in a sidebar.
+  const autogenerated_sidebar_directories = ["docs/rfc/active/"];
+  for (const filepath of markdown_files) {
+    if (
+      autogenerated_sidebar_directories.some((dir) => filepath.startsWith(dir))
+    ) {
+      // The sidebars for these directories is automatically generated,
+      // so we don't need check that they're in the sidebar.
+      continue;
+    }
+    if (!accounted_for_in_sidebar(filepath)) {
```

Comment:
we should just warn here

Reply:
If we just warn, I don't think anyone will ever see it 😦 

Given that most markdown files should probably make their way to the hosted docs site, I think it makes sense to error here.